File: suffixo_design_pattern/Singleton/LoggerSingleton.java
package suffixo_design_pattern.Singleton;

public class LoggerSingleton {
    private static LoggerSingleton instance;
    private String log = "";

    private LoggerSingleton() {}

    public static LoggerSingleton getInstance() {
        if (instance == null) {
            instance = new LoggerSingleton();
        }
        return instance;
    }

    public void addLog(String message) {
        log += message + "\n";
    }

    public String getLog() {
        return log;
    }
}

File: WILLYAM_GABIEL_DE_SOUZA_FONSECA/WILLYAM_GABRIEL_DE_SOUZA_FONSECA_design_pattern/QUESTAO_6/LoggerSingleton.js
// Singleton com estado global (má prática)
class LoggerSingleton {
  constructor() {
    if (LoggerSingleton.instance) {
      return LoggerSingleton.instance;
    }

    this.logs = [];
    LoggerSingleton.instance = this;
  }

  log(message) {
    const timestamp = new Date().toISOString();
    const fullMessage = `[${timestamp}] ${message}`;
    this.logs.push(fullMessage);
    console.log(fullMessage);
  }

  getLogs() {
    return this.logs;
  }
}
